v2/bundle/x509bundle/bundle_test.go

```diff
@@ -0,0 +1,216 @@
+package x509bundle
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_New(t *testing.T) {
+	bundle := New("example.org")
+	require.NotNil(t, bundle)
+	require.NotNil(t, bundle.X509Roots())
+	assert.Len(t, bundle.X509Roots(), 0)
+	assert.Equal(t, spiffeid.TrustDomain("example.org"), bundle.TrustDomain())
+}
+
+func Test_LoadSucceeds(t *testing.T) {
+	bundle, err := Load("example.org", "testdata/certs.pem")
+	require.NoError(t, err)
+	require.NotNil(t, bundle)
+	assert.Len(t, bundle.X509Roots(), 2)
+}
+
+func Test_LoadFails(t *testing.T) {
+	bundle, err := Load("example.org", "testdata/non-existent-file.pem")
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "unable to open file")
+	assert.Nil(t, bundle)
+}
+
+func Test_ReadSucceeds(t *testing.T) {
+	file, err := os.Open("testdata/certs.pem")
+	require.NoError(t, err)
+	defer file.Close()
```

Comment:
@azdagron I'm curious to know if this would be a place where the use of the newly supported cleanup functions is recommended? https://golang.org/pkg/testing/#T.Cleanup

Reply:
If this was a helper function that opened the file for testing, then Cleanup would be a good way of not making the caller have to arrange for cleanup. However, considering this is the test function itself, the traditional cleanup pattern using defer is just fine. Either way is appropriate.